merino/providers/weather/backends/accuweather.py

```diff
@@ -76,6 +76,17 @@ def __init__(
         self.url_current_conditions_path = url_current_conditions_path
         self.url_forecasts_path = url_forecasts_path
 
+    def cache_inputs_for_weather_report(self, geolocation: Location) -> Optional[bytes]:
+        """Return the inputs used to form the cache key for looking up and storing the current
+        conditions and forecast for a location.
+        """
+        if geolocation.country is None or geolocation.postal_code is None:
+            return None
+
+        return geolocation.country.encode("utf-8") + geolocation.postal_code.encode(
+            "utf-8"
+        )
```

Comment:
Perhaps overkill, but python also has [pattern matching](https://peps.python.org/pep-0622/)!
```suggestion
        match geolocation:
            case Location(country, postal_code) | if country is None or postal_code is None:
                 return None
             case Location(country, postal_code):
                 return f"{country}{postal_code}".encode("utf-8")
```

Reply:
Oooh, fancy—I like how that reads! 😄 Thanks!

Reply:
I tried writing this a few different ways—in addition to yours, this was one of the others:

```py
match (geolocation.country, geolocation.postal_code):
    case (None, _) | (_, None):
        return None
    case (country, postal_code):
        return f"{country}{postal_code}".encode("utf-8")
```

...But I couldn't figure out how to teach Mypy that `country` and `postal_code` are non-optional in the second arm! Even with various checks for `None` in the first arm, it still thinks both are optionals in the second arm, and also complains about a missing return statement—it doesn't see that the match is exhaustive.

Reply:
I found a few similar-but-not-quite-the-same reports in the Mypy issue tracker, and filed python/mypy#14731 for completeness.

Reply:
Of course, mypy ;)

We've seen this kind of false alarm with pattern matching before. Had to add a [dummy wildcard](https://github.com/mozilla-services/merino-py/blob/main/merino/utils/user_agent_parsing.py#L33) to appease mypy, yet also had to exclude that from coverage checking.

I will let you decide which route to take, feel free to stick to your original version. Sorry for taking you on this detour 🙇‍♂️

Reply:
That's okay! I'll go ahead and leave it as it is; with all the changes to make Mypy happy, this function ends up being twice as long. 😅